### ISSUE TITLE
Reproducibility audit: remaining estimators (follow-up to #1364, #1369, #1374, #1376)

### ISSUE THREAD

**Issue by immu4989** [microsoft/FLAML#1540]
## Background

@dannycg1996 has merged a series of reproducibility fixes for FLAML estimators:

- #1364 — CatBoost metrics reproducibility
- #1369 — LGBM reproducibility
- #1374 — ElasticNetEstimator reproducibility
- #1376 — LinearSVC reproducibility

These follow up on the long-standing reproducibility concern raised in #151.

## Proposal

I'd like to extend that work and audit the remaining estimators in
`flaml/automl/model.py` that accept a `random_state`. After reading the
merged PRs above to confirm the pattern, my plan is to submit one focused
PR per estimator (or per class family where inheritance makes a single PR
appropriate), each with:

- A reproducibility unit test in the style of the tests added in #1369 / #1374
- The minimal config / `__init__` / `config2params` change required to
  ensure a seeded run produces identical results across repeated fits

## Estimators not yet covered

| Estimator                    | Notes                                              |
|------------------------------|----------------------------------------------------|
| `RandomForestEstimator`      | Has fixed default `random_state` since v1.1.0;     |
|                              | want to verify end-to-end reproducibility          |
| `ExtraTreesEstimator`        | Inherits from `RandomForestEstimator`              |
| `XGBoostEstimator`           | Non-sklearn API path                               |
| `XGBoostSklearnEstimator`    | sklearn API path                                   |
| `XGBoostLimitDepthEstimator` | Inherits from `XGBoostSklearnEstimator`            |
| `LRL1Classifier`             | LR with L1 — relevant for `liblinear`/`saga`       |
| `LRL2Classifier`             | LR with L2 — relevant for `sag`/`saga`             |
| `SGDEstimator`               | Stochastic gradient descent                        |

## Questions for maintainers

1. Is this audit welcome, and is the per-estimator-PR approach preferred
   over a single combined PR?
2. Should the audit also cover the corresponding "flamlized" zero-shot
   estimators in `flaml/default/estimator.py`, or are those out of scope?
3. Are there estimators on the list above that you'd like me to skip
   (e.g., known-deprecated paths)?

I'd plan to start with `RandomForestEstimator` once you confirm the
direction. Happy to adjust scope based on your guidance.

### Additional context

##Why this matters to me

In production ML for HR risk modeling, reproducibility is a regulatory
and audit requirement — being able to re-derive the same model from the
same data and config is non-negotiable. FLAML is increasingly used in
those settings (including via Microsoft Fabric), so closing this gap
across the estimator family seems worthwhile.

**Comment by thinkall** [comment]
Hi @immu4989 ,thank you very much for the proposal. PRs are super welcome. I think one PR for each estimator is better than putting everything in a single PR.

**Comment by immu4989** [comment]
Sure @thinkall . Thank you for the information. Submitted a pull request for **SGD Estimator**.

**Comment by immu4989** [comment]
Closing this out — the audit is complete. Six PRs landed over the past three weeks, covering all 8 estimators originally listed:

| PR | Estimator(s) | Notes |
|---|---|---|
| #1541 | `SGDEstimator` | First in the series; established the `random_seed`-aware `__init__` pattern |
| #1546 | `LRL1Classifier` | Uncommented `# "lrl1"` from the test parametrize lists at the same time |
| #1547 | `RandomForestEstimator`, `ExtraTreesEstimator` | One PR, two estimators (inheritance) |
| #1549 | `XGBoostSklearnEstimator`, `XGBoostLimitDepthEstimator` | One PR, two estimators (inheritance) |
| #1551 | `XGBoostEstimator` | Non-sklearn `xgb.train(...)` code path |
| #1552 | `LRL2Classifier` | Forward-safety for `sag` / `saga` solver overrides; final estimator in the family |

All PRs follow the same pattern in `__init__`:

```python
random_seed = self.params.pop("random_seed", config.get("random_seed", 10242048))
if "random_state" not in self.params:
    self.params["random_state"] = random_seed
```
RandomForestEstimator retained its existing 12032022 default for back compat; everything else uses 10242048 per @dannycg1996's earlier convention.



Thanks @thinkall for the consistent reviews throughout — appreciated the steady cadence on these.

Next up I'd like to look at #1200 (SMOTE with cross validation), which is a related quality of life ask for imbalanced classification workflows. Will open a separate discussion if/when I have a concrete proposal.
